public/app/plugins/datasource/cloud-monitoring/plugin.json

```diff
@@ -3,6 +3,7 @@
   "type": "datasource",
   "id": "stackdriver",
   "category": "cloud",
+  "builtIn": true,
```

Comment:
perhaps unrelated to this PR... 

Is this something we actually need/want to set manually?  it should only ever be set to true for the actually builtin plugins, and must be false elsewhere.  As long as we assert that and never believe what is in the raw file, then we are good

Reply:
The declaration in the schema:

```cue
builtIn: bool | *false
```

means that it at least needn't be specified in external plugins - they'll default to `false`, as appropriate. We'd probably then need to take additional steps to reject publishing any plugins where it's set to `true`. Here's one case, with `id`:

https://github.com/grafana/grafana/pull/53754/files#diff-73442850ef4f96720633490ee778b299131e87b09488cd109a8162fe2184e877R15-R17

This was basically the simplest path i had available to fixing validation errors that were popping up. i have no strong stance on how we tackle this - only that it's quite valuable to reuse the same logic for core and external plugins.

Reply:
I wonder how much effort it would involve to make core plugins conform instead - that would be my preference and aligns better with the potential goal of removing their snowflakey-ness. Can we run them through the schema validation and see all fields that complain?

Reply:
> Can we run them through the schema validation and see all fields that complain?

Happily, this PR does that!

This PR both [introduces the `pfs.ParsePluginFS`](https://github.com/grafana/grafana/pull/53754/files#diff-2910f3982292a1758b4b3aca45ea9e7f032773018451d91e2f8f63323146ebaaR156) func, which runs schema-based validation on everything you pass it, and [rebuilds the code generator](https://github.com/grafana/grafana/pull/53754/files#diff-63ed6c0c1d3b6c1b6fc705eb25d8bbbb1e8834b01ea52939d0410feaf392b50dR51) so that codegen [depends on the new validating parser](https://github.com/grafana/grafana/pull/53754/files#diff-fe41506a5c1349d899b28c4c8e4e6c1852ebcd96c27e0be42ec54e8376b5276eR75). So, because codegen is running across all core plugins, and codegen depends on the validating parser, core plugins' `plugin.json` is all fixed up :)

That's why i changed a bunch of them. Not terribly big changes, only took about half an hour to actually do. There were a couple that don't validate [which i excluded](https://github.com/grafana/grafana/pull/53754/files#diff-63ed6c0c1d3b6c1b6fc705eb25d8bbbb1e8834b01ea52939d0410feaf392b50dR28-R30), but i made a comment of where the problem is. It's easy to fix in a follow-up - change the schema, drop them from the list, see if `make gen-cue` works.